File: Demo/app/src/main/java/com/china/bosh/demo/ui/activity/BaseActivity.java
package com.china.bosh.demo.ui.activity;

import android.app.Activity;
import android.app.Application;
import android.content.ComponentCallbacks;
import android.content.res.Configuration;
import android.os.Bundle;
import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.util.DisplayMetrics;

import com.china.bosh.demo.application.MyApplication;
import com.china.bosh.mylibrary.utils.PermissionsManager;
import com.china.bosh.mylibrary.utils.PermissionsResultAction;

import butterknife.ButterKnife;

/**
 * @author lzq
 * @date 2018/9/6
 */

public abstract class BaseActivity extends com.china.bosh.mylibrary.ui.activity.BaseActivity{

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: kotlin_module/module_kotlin/src/main/java/com/bosh/module_kotlin/BaseActivity.kt
package com.bosh.module_kotlin

import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity

/**
 * @author lzq
 * @date  2019-10-28
 */
abstract class BaseActivity : AppCompatActivity() {

    val TAG = javaClass.simpleName


    protected abstract fun getLayoutRes() :  Int

    protected abstract fun initData()

    protected abstract fun initView()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.e(TAG, "onCreate")
        setContentView(getLayoutRes())
        initData()
        initView()
    }

    fun toast(msg: String?) {
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show()
    }
}